TundraTD/Assets/Scripts/Spells/SpellClasses/LightningCrystallSpell.cs

```diff
@@ -1,11 +1,56 @@
 ﻿using Spells;
+using System.Collections.Generic;
+using System.Linq;
 using UnityEngine;
 
 public class LightningCrystallSpell : MagicSpell
 {
+    private List<Rigidbody> _mobs = new List<Rigidbody>();
+    private float t = 0;
+
+    [SerializeField] float pullForce;
+    [SerializeField] float timeToLive;
     public override BasicElement Element => BasicElement.Earth | BasicElement.Lightning;
 
+    private void OnTriggerEnter(Collider e)
+    {
+        if (!e.gameObject.TryGetComponent(out Rigidbody enteredMob))
+            return;
+        _mobs.Add(enteredMob);
+    }
+
+    private void OnTriggerExit(Collider e)
+    {
+        if (!e.gameObject.TryGetComponent(out Rigidbody enteredMob))
+            return;
+        _mobs.Remove(enteredMob);
+    }
+
+    private void FixedUpdate()
+    {
+        _mobs.Where(mob => !mob.gameObject.activeSelf).ToList().ForEach(mob => _mobs.Remove(mob));
```

Comment:
Насколько я понял, эта строчка убирает из списка мобов тех, что уже успели умереть. Я бы предпочел избежать в FixedUpdate() приведения данных к списку, так как это тяжелая операция. Мне кажется лучше будет просто проходиться через for i (...) и внутри проверять на activeself. Тогда у нас в целом будет один цикл for для обоих итераций

```suggestion
        for (int i = 0; i < _mobs.Length; i++)
        {
            var mob = _mobs[i];
            if (!mob.gameObject.activeSelf)
            {
                _mobs.RemoveAt(i);
            }
            else 
            {
            var mobPosition = mob.transform.position;
              mob.AddForce(new Vector3 (mobPosition .x - transform.position.x, mobPosition.y - transform.position.y, mobPosition.z - transform.position.z).normalized * pullForce * -1, ForceMode.Force));
            }
        }
```

Reply:
А ещё, я думаю, что мы можем использовать для этого эффект Graviation, который будет принимать Vector3 targetPosition, к которому будет притягиваться моб. Но возможно, это излишнее